service/pom.xml

```diff
@@ -291,13 +291,6 @@
       <scope>test</scope>
     </dependency>
 
-    <dependency>
-      <groupId>org.apache.directory.client.ldap</groupId>
-      <artifactId>ldap-client-api</artifactId>
-      <version>${apache-directory-clientapi.version}</version>
-      <scope>test</scope>
-    </dependency>
-
     <dependency>
       <groupId>org.apache.directory.server</groupId>
       <artifactId>apacheds-server-integ</artifactId>
```

Comment:
Can you clarify why we don't want to `exclude ldap-client-api` from here as it was done in the original commit in master (https://github.com/apache/hive/commit/5581eb8a74a4f33b35b7bf70d9ec4e9a95f3b8a0).

When I run `mvn dependency:tree` in service module I see the `org.apache.directory.client.ldap:ldap-client-api` dependency coming transitively.
```
[INFO] +- org.apache.directory.server:apacheds-server-integ:jar:1.5.7:test
[INFO] |  +- org.apache.directory.server:apacheds-interceptor-kerberos:jar:1.5.7:test
[INFO] |  |  +- org.apache.directory.server:apacheds-core:jar:1.5.7:test
[INFO] |  |  |  +- org.apache.directory.server:apacheds-core-api:jar:1.5.7:test
[INFO] |  |  |  |  +- org.apache.directory.server:apacheds-core-entry:jar:1.5.7:test
[INFO] |  |  |  |  \- org.apache.directory.server:apacheds-core-constants:jar:1.5.7:test
[INFO] |  |  |  +- org.apache.directory.server:apacheds-utils:jar:1.5.7:test
[INFO] |  |  |  \- bouncycastle:bcprov-jdk15:jar:140:test
[INFO] |  |  \- org.apache.directory.server:apacheds-kerberos-shared:jar:1.5.7:test
[INFO] |  |     +- org.apache.directory.server:apacheds-core-jndi:jar:1.5.7:test
[INFO] |  |     \- org.apache.directory.server:apacheds-protocol-shared:jar:1.5.7:test
[INFO] |  +- org.apache.directory.server:apacheds-core-integ:jar:1.5.7:test
[INFO] |  +- ldapsdk:ldapsdk:jar:4.1:test
[INFO] |  +- org.apache.directory.client.ldap:ldap-client-api:jar:0.1:test
```

Reply:
Commit 5581eb8a74a4f33b35b7bf70d9ec4e9a95f3b8a0 actually removed the exclusion from the existing codebase. This exclusion is not present in branch-3, so there isn't anything for this patch to remove.

If I run `mvn dependency:tree` on master, I see the same transitive dependency. Looking back on the history of [HIVE-21836](https://issues.apache.org/jira/browse/HIVE-21836), I see there was a Jenkins run for HIVE-21836.01.patch, which did not remove the exclusion, and it had a test failure in `TestLdapAtnProviderWithMiniDS`. Then, HIVE-21836.02.patch removed the exclusion, and it got a successful Jenkins run. I can't reproduce that problem though. I tried adding the exclusion back, and the test still passed fine for me.

Let me know if you think we should bring back this exclusion on both master and branch-3. Thanks!

Reply:
Oh you are right. I was confused while checking the diff and though that the exclusion was added rather than removed. Apologies for the back and forth!